sh/helpers_test.go

```diff
@@ -0,0 +1,140 @@
+package sh_test
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/magefile/mage/sh"
+)
+
+func makeTestFile(dir string, name string) (string, error) {
```

Comment:
so you can probably replace this with [ioutil.WriteFile](https://golang.org/pkg/io/ioutil/#WriteFile).  I don't think the random contents are critical to the test.

Reply:
I think I disagree with that. I can imagine situations where copy was implemented using a fixed size buffer and a fencepost error caused corruption for specific file sizes.

Reply:
I think that if we want to do a bunch of tests with different contents, that's fine, but doing one test with random contents means that we won't know what caused the problem if the test fails.  Also, we're using io.Copy, so I'm not too worried about the actual copying of contents being wrong :)

Reply:
OK, if you're happy trusting io.Copy fair enough.

Reply:
if io.Copy was broken, someone would have figured it out long before us :)